sui-execution/latest/sui-verifier/src/private_generics.rs

```diff
@@ -102,14 +107,21 @@ fn verify_private_transfer(
     view: &BinaryIndexedView,
     fhandle: &FunctionHandle,
     type_arguments: &[SignatureToken],
+    protocol_version: u64,
 ) -> Result<(), String> {
+    let public_transfer_functions = if protocol_version >= 33 {
```

Comment:
Instead of making the verifier aware of protocol versions, can we add a flag to `VerifierConfig` to control this behaviour?  That was the intended way to make protocol-directed behaviour changes here.

Reply:
Thanks for the suggest @amnn! Updated to use the `VerifierConfig` instead of the protocol version directly -- the changes from the original are in the top commit on this PR.